src/into_url.rs

```diff
@@ -8,6 +10,7 @@ pub trait IntoUrl: IntoUrlSealed {}
 
 impl IntoUrl for Url {}
 impl IntoUrl for String {}
+impl<'a> IntoUrl for Cow<'a, str> {}
```

Comment:
Hi, I don't get the motivation for reducing heap allocation here. Even when we have an owned `String`, we're still calling `Url::parse(&str)` to parse it. 

It looks like we don't need to add impl for `Cow<'a, str>`; Maybe we could just store a `Cow<'a, str>` and use `as_ref()` to obtain a `&str` for parsing?

Reply:
so after impl IntoUrl for Cow, I and do this
```
pub fn resolve_proxy_from_url(target_url: &Url) -> Option<Cow<'static, str>> {
    if let Some(match_env) = ENV_PROXY_CACHE.get(target_url.scheme()) {
        return Some(Cow::Borrowed(match_env.as_str()));
    }
    #[cfg(any(target_os = "windows", target_os = "macos"))]
    {
        let system_proxy_cache = get_proxy_cache();
        if let Some(match_system) = system_proxy_cache.get(target_url.scheme()) {
            return Some(Cow::Owned(match_system.into()));
        }
    }
    None
}
```
does that make sense?  @Xuanwo

Reply:
Do you think it's a good idea to just return `Url` here? Regardless of whether we return `&str` or `String`, we'll still need to convert it into an owned `Url`. If we handle this conversion ourselves, we can avoid making an unnecessary copy of the `String`.

Reply:
> Do you think it's a good idea to just return `Url` here? Regardless of whether we return `&str` or `String`, we'll still need to convert it into an owned `Url`. If we handle this conversion ourselves, we can avoid making an unnecessary copy of the `String`.

@Xuanwo Oh right....thanks, you are right, just wrapped my head around with your suggestion, Didn't notice that Url itself impl IntoUrl as well, I can just return a Url directly here, thanks for your review, that's really helpful